Sources/MapboxCoreNavigation/Distance/RoadObjectDistanceInfo.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+import CoreLocation
+import MapboxNavigationNative
+
+/**
+ * Contains information about distance to the road object of a concrete
+ * type/shape (gantry, polygon, line, point etc.).
+ */
+public enum RoadObjectDistanceInfo {
+    case point(PointDistanceInfo)
+    case gantry(GantryDistanceInfo)
+    case polygone(PolygonDistanceInfo)
+    case subgraph(SubgraphDistanceInfo)
+    case line(LineDistanceInfo)
```

Comment:
For simplicity, can we inline each of these “Info” types as individual associated values on these cases? It looks like these types are only used here.